dotnet/src/SemanticKernel.Abstractions/Kernel.cs

```diff
@@ -444,7 +444,35 @@ public CultureInfo Culture
     {
         Verify.NotNull(function);
 
-        return function.InvokeStreamingAsync<StreamingContentBase>(this, arguments, CancellationToken.None);
+        return function.InvokeStreamingAsync<StreamingContentBase>(this, arguments, cancellationToken);
+    }
+
+    /// <summary>
+    /// Invokes the <see cref="KernelFunction"/> and streams its results.
+    /// </summary>
+    /// <param name="pluginName">The name of the plugin containing the function to invoke. If null, all plugins will be searched for the first function of the specified name.</param>
+    /// <param name="functionName">The name of the function to invoke.</param>
+    /// <param name="arguments">The arguments to pass to the function's invocation, including any <see cref="PromptExecutionSettings"/>.</param>
+    /// <param name="cancellationToken">The <see cref="CancellationToken"/> to monitor for cancellation requests. The default is <see cref="CancellationToken.None"/>.</param>
+    /// <returns>An <see cref="IAsyncEnumerable{T}"/> for streaming the results of the function's invocation.</returns>
+    /// <exception cref="ArgumentNullException"><paramref name="functionName"/> is null.</exception>
+    /// <exception cref="ArgumentException"><paramref name="functionName"/> is composed entirely of whitespace.</exception>
+    /// <exception cref="KernelFunctionCanceledException">The <see cref="KernelFunction"/>'s invocation was canceled.</exception>
+    /// <remarks>
+    /// The function will not be invoked until an enumerator is retrieved from the returned <see cref="IAsyncEnumerable{T}"/>
+    /// and its iteration initiated via an initial call to <see cref="IAsyncEnumerator{T}.MoveNextAsync"/>.
+    /// </remarks>
+    public IAsyncEnumerable<StreamingContentBase> InvokeStreamingAsync(
```

Comment:
nit: Since `InvokeStreamingAsync` is one of the main entry points in Kernel and users will interact with result type a lot, I'm wondering if it makes sense to rename `StreamingContentBase` to just `StreamingContent`. 

Instead of this:
```csharp
await foreach (StreamingContentBase chunk in kernel.InvokeStreamingAsync(function))
{
   // Process streaming...
}
```

Users will have this:
```csharp
await foreach (StreamingContent chunk in kernel.InvokeStreamingAsync(function))
{
   // Process streaming...
}
```

Another way would be to make it consistent with `Task<FunctionResult> InvokeAsync` and for streaming have `IAsyncEnumerable<FunctionStreamingResult> InvokeStreamingAsync`.

Reply:
@dmytrostruk It was `StreamingContent` before and the analogous for non-streaming was `ModelContent`, since `ModelContent` was not a good name `ContentBase` was used, but I agree, when thinking about `StreamingContentBase` we need to also think about `ContentBase`, any change there should reflect in non-streaming as well.

Reply:
> any change there should reflect in non-streaming as well

That's not always the case, because if we look from usability perspective, `StreamingContentBase` class will be used more than `ContentBase` and it will be involved in main scenarios (`kernel.InvokeStreamingAsync`), so the importance of having a good name for `StreamingContentBase` is bigger, comparing to `ContentBase`. While `ContentBase` could be used to create concrete content like `TextContent : ContentBase`, `ImageContent: ContentBase` etc, so maybe in this case `ContentBase` naming makes sense. So, my point is that it's possible to trade-off consistency in naming hierarchy for better usability. 

But ideal scenario would be to have both consistent and usable naming :) In this case, maybe we can consider having `KernelStreamingContent` and `KernelContent` models or something similar.

Reply:
Resolving this comment, as I'm working on naming and namespaces in scope of separate PR.